### ISSUE TITLE
New clusterRoleName helper seems to be causing a template error

### ISSUE THREAD

**Issue by nikki-quant** [coredns/helm#180]
Hi folks, thanks for maintaining CoreDNS!

I'm looking to update from version 1.30.0 to 1.36.0 of the CoreDNS chart.

When I run helm template against the new version without making any changes to values files I get the following error:

```	
		STDERR:
		  Error: template: coredns/templates/_helpers.tpl:233:38: executing "coredns.clusterRoleName" at <.Values.clusterRole.nameOverride>: nil pointer evaluating interface {}.nameOverride
		  Use --debug flag to render out invalid YAML
```

This looks to be caused by the following snippet in the helper templates for the chart:

```
{{/*
Create the name of the service account to use
*/}}
{{- define "coredns.clusterRoleName" -}}
{{- if and .Values.clusterRole .Values.clusterRole.nameOverride -}}
    {{ .Values.clusterRole.nameOverride }}
{{- else -}}
    {{ template "coredns.fullname" . }}
{{- end -}}
{{- end -}}
```

Without the `clusterRole` key set in the values, Helm errors when trying to evaluate whether `.Values.clusterRole.nameOverride` is defined, because it is trying to access a child property of a parent which does not exist. As far as I understand, this is the expected behaviour in Helm and their recommendation is to prefer using relatively flat variable names: https://github.com/helm/helm/issues/8026#issuecomment-677851638)

Adding an empty `clusterRole: {}` key in a values file lets me template the chart successfully.

As best I can tell, `clusterRole` was currently only used as a parent for the new `clusterRole.nameOverride`. Should it be added to the chart default values as an empty dictionary to avoid running into this templating error? ie:

```
# /charts/coredns/values.yaml

clusterRole: {}
```

Thanks,
Nikki

**Comment by hagaibarel** [comment]
Hi @nikki-quant , thanks for raising this.

I think you're correct and the lack of a default value causes a nil pointer exception. The right value should be an empty string  (`""`) as we expect a string vaule there not object.

I would be happy to review a PR with the fix

**Comment by nikki-quant** [comment]
Thank for taking a look -  I'll put a PR in :)

**Comment by nikki-quant** [comment]
Hi @hagaibarel , I've put in a pull request. Very happy to make updates if that doesn't look right, or if there's styleguide type things you'd prefer done differently for the changelog or comments.

**Comment by hagaibarel** [comment]
Fixed in #181
